src/preview.ts

```diff
@@ -31,20 +31,31 @@ const renderToCanvas: RenderToCanvas<RRWebFramework> = async (context, element)
   const response = await fetch(`${url}/${id}`);
   const snapshot = await response.json();
 
+  // The snapshot is a representation of a complete HTML document. The first child is the
+  // doc type declaration and the second is the html element.
   const htmlNode = snapshot.childNodes[1];
 
+  // If you rebuild the full snapshot with rrweb (the document) it will replace the
+  // current document and call `document.open()` in the process, which unbinds all event handlers
+  // (and breaks Storybook).
+  // However, if you just rebuild the html element part, it will recreate but not attempt to
+  // insert it in the DOM.
   // @ts-expect-error rebuild is typed incorreclty, cache and mirror are optional
   const html = (await rebuild(htmlNode, { doc: document })) as HTMLElement;
 
-  document.children[0].innerHTML = html.innerHTML;
+  // Now we insert the rebuilt html element in the DOM
+  document.replaceChild(html, document.children[0]);
 
-  // insert a couple of elements to fool SB
-  document.body.innerHTML += '<div id="storybook-root"></div><div id="storybook-docs"></div>';
+  // Storybook's WebView will throw an error if it cannot find these two ids in the DOM.
+  // We never render docs (so the #storybook-docs doesn't matter), and our`renderToCanvas`
+  // function is already ignoring the #storybook-root (`element` above), so it doesn't matter where
+  // they are or what they contain.
+  // We make them a script in the head to ensure they don't impact layout.
+  document.head.innerHTML +=
```

Comment:
Are we guaranteed to have a `head`? 🤔

Reply:
🤔 --- I ...think... so? I don't really know how to say so for sure, but I've never seen an HTML doc without one :)

Reply:
I tried this in a browser, does that satisfy you @tevanoff ? :)

<img width="505" alt="image" src="https://github.com/chromaui/archive-storybook/assets/132554/6986d122-29cf-43f8-b114-eb1bb153b800">

Reply:
The head will likely be inserted into the document if it's not specified, because the `<title>` section is required (at least by HTML 3.0, and I don't _think_ this changed in HTML 5). Thus, the browser will likely insert a header section into the DOM if it's not defined.